src/topology/config/mod.rs

```diff
@@ -205,6 +205,29 @@ pub trait TransformConfig: core::fmt::Debug {
     fn transform_type(&self) -> &'static str;
 }
 
+#[derive(Debug, Clone)]
+pub struct TransformContext {
+    pub(super) exec: TaskExecutor,
+    pub(super) resolver: Resolver,
+}
+
+impl TransformContext {
+    pub fn new_test(exec: TaskExecutor) -> Self {
+        Self {
+            resolver: Resolver::new(Vec::new(), exec.clone()).unwrap(),
+            exec,
+        }
+    }
+
+    pub fn exec(&self) -> &TaskExecutor {
```

Comment:
nit: can we rename this to `executor`, I prefer the fully qualified name for fn but exec is fine for parameters.